lib/membrane_wav/postprocessing.ex

```diff
@@ -3,40 +3,76 @@ defmodule Membrane.WAV.Postprocessing do
   Module responsible for post-processing serialized WAV files.
 
   Due to the fact that `Membrane.WAV.Serializer` creates WAV file with incorrect `file length` and
-  `data length` blocks in the header, post-processing is needed. `correct_wav_header/1` fixes that
+  `data length` blocks in the header, post-processing is needed. `fix_wav_header/1` fixes that
   problem.
 
   Header description can be found in `Membrane.WAV.Parser`.
   """
 
+  @type fix_error :: {:error, File.posix() | :invalid_file | :badarg | :terminated} | IO.nodata()
+
   @doc """
   Fixes header of the WAV file located in `path`.
   """
-  @spec correct_wav_header(String.t()) :: :ok
-  def correct_wav_header(path) do
-    with {:ok, info} <- File.stat(path),
-         {:ok, file} <- File.open(path, [:binary, :read, :write]) do
-      file_length = info.size
-      header_length = get_header_length(file, 0)
-      data_length = file_length - header_length
+  @spec fix_wav_header(String.t()) :: :ok | fix_error()
+  def fix_wav_header(path) do
+    with {:ok, file_length} <- get_file_length(path),
+         {:ok, file} <- File.open(path, [:binary, :read, :write]),
+         {:ok, header_length} <- get_header_length(file),
+         data_length <- file_length - header_length,
+         :ok <- update_file(file, file_length, header_length, data_length),
+         :ok <- File.close(file) do
+      :ok
+    end
+  end
 
-      :file.position(file, 4)
-      IO.binwrite(file, <<file_length - 8::32-little>>)
+  defp get_file_length(path) do
+    with {:ok, info} <- File.stat(path) do
+      {:ok, info.size}
+    end
+  end
 
-      :file.position(file, header_length - 4)
-      IO.binwrite(file, <<data_length::32-little>>)
+  defp get_header_length(file) do
+    with "RIFF" <- IO.binread(file, 4),
+         {:ok, _new_position} <- :file.position(file, 8),
+         <<"WAVE", "fmt ", format_chunk_size::32-little>> <- IO.binread(file, 12),
+         current_position <- 20 + format_chunk_size,
```

Comment:
That's non-failing assignment, usually a regular `=` is used then:
```suggestion
         current_position = 20 + format_chunk_size,
```